examples_tests/21.DynamicTextureIndexing/main.cpp

```diff
@@ -1,319 +1,383 @@
+            ICPUImageView* texBump = dynamic_cast<ICPUImageView*>(ds3->getDescriptors(6u).begin()->desc.get());
+            assert(tex);
+            assert(texBump);
+
+            auto a = textures.insert(std::make_pair(tex, texBinding));
+            if (a.second == true)
+                texBinding++;
+
+            auto b = textures.insert(std::make_pair(texBump, texBinding));
+            if (b.second == true)
+                texBinding++;
+
+            if (texBinding >= 15u)
+            {
+                mbRangesTex.push_back({ {rangeBegin, it + 1}, textures });
+                rangeBegin = it + 1;
+                texBinding = 0;
+                textures.clear();
+            }
+
+            if (it + 1 == meshBuffers.end())
+            {
+                mbRangesTex.push_back({ {rangeBegin, meshBuffers.end()}, textures });
+                break;
+            }
+        }
+    }
+    
+    struct CustomIndirectCommand : DrawElementsIndirectCommand_t
+    {
+        uint32_t diffuseTexBinding = std::numeric_limits<uint32_t>::max();
+        uint32_t bumpTexBinding = std::numeric_limits<uint32_t>::max();
+    };
+
+    struct DrawIndexedIndirectInput
+    {
+        asset::SBufferBinding<video::IGPUBuffer> vtxBindings[video::IGPUMeshBuffer::MAX_ATTR_BUF_BINDING_COUNT];
+        asset::E_PRIMITIVE_TOPOLOGY mode = asset::EPT_TRIANGLE_LIST;
+        asset::E_INDEX_TYPE indexType = asset::EIT_16BIT;
+        core::smart_refctd_ptr<video::IGPUBuffer> indexBuff = nullptr;
+        core::smart_refctd_ptr<video::IGPUBuffer> indirectDrawBuff = nullptr;
+        size_t offset = 0u;
+        size_t maxCount = 0u;
+        size_t stride = 0u;
+        core::smart_refctd_ptr<video::IGPUBuffer> countBuffer = nullptr;
+        size_t countOffset = 0u;
+    };
+
+    core::vector<asset::MeshPackerBase::PackedMeshBuffer<asset::ICPUBuffer>> packedMeshBuffer(mbRangesTex.size());
+    core::vector<DrawIndexedIndirectInput> mdiCallParams(mbRangesTex.size());
+    core::vector<core::smart_refctd_ptr<IGPUBuffer>> gpuIndirectDrawBuffer(mbRangesTex.size());
+    for(size_t i = 0u; i < mbRangesTex.size(); i++)
+    {
+        asset::SVertexInputParams& inputParams = mesh_raw->getMeshBuffer(0u)->getPipeline()->getVertexInputParams();
+        asset::MeshPackerBase::AllocationParams allocationParams;
+
+        auto& mbRange = mbRangesTex[i].mbRanges;
+        core::vector<asset::MeshPackerBase::PackedMeshBufferData> pmbData;
+        const ptrdiff_t meshBuffersInRangeCnt = std::distance(mbRange.begin(), mbRange.end());
+
+        allocationParams.indexBuffSupportedCnt = 20000000u;
+        allocationParams.indexBufferMinAllocSize = 5000u;
+        allocationParams.vertexBuffSupportedCnt = 20000000u;
+        allocationParams.vertexBufferMinAllocSize = 5000u;
+        allocationParams.MDIDataBuffSupportedCnt = 20000u;
+        allocationParams.MDIDataBuffMinAllocSize = 1u; //so structs are adjacent in memory
```

Comment:
they should be adjacent no matter the min alloc size (if you `alloc` and `commit` the whole list of meshes that you wish to be adjacent)